Fail to build ARC zephyr with MetaWare toolchain

**Describe the bug**
A clear and concise description of what the bug is.
What have you tried to diagnose or workaround this issue?

**To Reproduce**
Steps to reproduce the behavior:
1. west build -b nsim_em -d build tests/kernel/common
2. See error



**Logs and console output**
```
west build -b nsim_em -d build tests/kernel/common/
-- west build: generating a build system
Including boilerplate (Zephyr base): /SCRATCH2/Watson/zephyrproject/zephyr/cmake/app/boilerplate.cmake
-- Application: /SCRATCH2/Watson/zephyrproject/zephyr/tests/kernel/common
-- Zephyr version: 2.5.99 (/SCRATCH2/Watson/zephyrproject/zephyr)
-- Found Python3: /depot/Python/Python-3.7.0/bin/python3.7 (found suitable exact version "3.7.0") found components: Interpreter 
-- Found west (found suitable version "0.7.2", minimum required is "0.7.1")
-- Board: nsim_em
-- Cache files will be written to: /u/zhiwei/.cache/zephyr
-- Using toolchain: zephyr 0.12.3 (/SCRATCH2/svc-arcoss_auto/tools/zephyr-sdk/0.12.3)
-- Found dtc: /SCRATCH2/svc-arcoss_auto/tools/zephyr-sdk/0.12.3/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.5.0", minimum required is "1.4.6")
-- Found BOARD.dts: /SCRATCH2/Watson/zephyrproject/zephyr/boards/arc/nsim/nsim_em.dts
-- Generated zephyr.dts: /SCRATCH2/Watson/zephyrproject/zephyr/build/zephyr/zephyr.dts
-- Generated devicetree_unfixed.h: /SCRATCH2/Watson/zephyrproject/zephyr/build/zephyr/include/generated/devicetree_unfixed.h
-- Generated device_extern.h: /SCRATCH2/Watson/zephyrproject/zephyr/build/zephyr/include/generated/device_extern.h
Parsing /SCRATCH2/Watson/zephyrproject/zephyr/Kconfig
Loaded configuration '/SCRATCH2/Watson/zephyrproject/zephyr/boards/arc/nsim/nsim_em_defconfig'
Merged configuration '/SCRATCH2/Watson/zephyrproject/zephyr/tests/kernel/common/prj.conf'
Configuration saved to '/SCRATCH2/Watson/zephyrproject/zephyr/build/zephyr/.config'
Kconfig header saved to '/SCRATCH2/Watson/zephyrproject/zephyr/build/zephyr/include/generated/autoconf.h'
-- The C compiler identification is Clang 10.0.1
-- The CXX compiler identification is Clang 10.0.1
-- The ASM compiler identification is unknown
-- Found assembler: /global/apps/mwdt_2020.12/MetaWare/arc/bin/ccac
-- Warning: Did not find file Compiler/-ASM
CMake Warning at ../../../cmake/linker/arcmwdt/target.cmake:137 (message):
  MWDT toolchain does not support

             CONFIG_LINKER_ORPHAN_SECTION_WARN
Call Stack (most recent call first):
  ../../../CMakeLists.txt:242 (toolchain_ld_baremetal)


-- Configuring done
-- Generating done
-- Build files have been written to: /SCRATCH2/Watson/zephyrproject/zephyr/build
-- west build: building application
[1/158] Preparing syscall dependency handling

[141/158] Linking C executable zephyr/app_smem_unaligned_prebuilt.elf
FAILED: zephyr/app_smem_unaligned_prebuilt.elf zephyr/app_smem_unaligned_prebuilt.map 
: && /global/apps/mwdt_2020.12/MetaWare/arc/bin/ccac    zephyr/CMakeFiles/app_smem_unaligned_prebuilt.dir/misc/empty_file.c.obj  -o zephyr/app_smem_unaligned_prebuilt.elf  -Wl,-T/SCRATCH2/Watson/zephyrproject/zephyr/build/zephyr/linker_app_smem_unaligned.cmd  -Wl,--gc-sections  -Wl,--entry=__start  -Wl,--Map=/SCRATCH2/Watson/zephyrproject/zephyr/build/zephyr/app_smem_unaligned_prebuilt.map  -Wl,--whole-archive  app/libapp.a  zephyr/libzephyr.a  zephyr/arch/common/libarch__common.a  zephyr/arch/arch/arc/core/libarch__arc__core.a  zephyr/arch/arch/arc/core/mpu/libarch__arc__core__mpu.a  zephyr/lib/libc/minimal/liblib__libc__minimal.a  zephyr/lib/posix/liblib__posix.a  zephyr/subsys/testsuite/ztest/libsubsys__testsuite__ztest.a  zephyr/drivers/serial/libdrivers__serial.a  -Wl,--no-whole-archive  zephyr/kernel/libkernel.a  zephyr/CMakeFiles/offsets.dir/./arch/arc/core/offsets/offsets.c.obj  -L/SCRATCH2/Watson/zephyrproject/zephyr/build/zephyr  zephyr/arch/common/libisr_tables.a  -Wl,--sort-section=alignment  -Wl,-u_OffsetAbsSyms  -Wl,-u_ConfigAbsSyms  -Hlld  -Hnosdata  -Hnocrt  -Xtimer0  -Hnoxcheck_obj  -Hnocplus  -Hcl  -Hheap=0  -Hnoivt && cd /SCRATCH2/Watson/zephyrproject/zephyr/build/zephyr && /depotbld/RHEL7.0/Python/Python-3.7.0/lib/python3.7/site-packages/cmake/data/bin/cmake -E echo
lldac: error: /SCRATCH2/Watson/zephyrproject/zephyr/build/zephyr/linker_app_smem_unaligned.cmd:216: ) expected, but got ,
>>>  } GROUP_ROM_LINK_IN(DCCM, ICCM)
>>>                          ^
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /depot/Python/Python-3.7.0/bin/cmake --build /SCRATCH2/Watson/zephyrproject/zephyr/build
```

**Environment (please complete the following information):**
 - OS: (Linux)
 - Toolchain (mwdt/2020.12)
 - Commit SHA ([6ca6b22ac](https://github.com/zephyrproject-rtos/zephyr/commit/6ca6b22ac755726ed541cccbf8bef9409fc8831a))


**Additional context**

This issue is introduced by this commit: [linker-tool-gcc: revise for MMU support](https://github.com/zephyrproject-rtos/zephyr/commit/acda9bf9cec8cc366e5a319f7f35cc234a6e34d3)

--- a/include/linker/intlist.ld
+++ b/include/linker/intlist.ld
@@ -34,8 +34,8 @@
 #ifndef LINKER_PASS2
 SECTION_PROLOGUE(.intList,,)
 {
-	KEEP(*(.irq_info))
-	KEEP(*(.intList))
+	KEEP(*(.irq_info*))
+	KEEP(*(.intList*))
 } GROUP_ROM_LINK_IN(IDT_LIST, IDT_LIST)
 #else
 /DISCARD/ :

--- a/include/linker/linker-tool-mwdt.h
+++ b/include/linker/linker-tool-mwdt.h
@@ -46,6 +46,14 @@
  */
 #define GROUP_LINK_IN(where) > where
 
+/**
+ * The GROUP_ROM_LINK_IN() macro is located at the end of the section
+ * description and tells the linker that this a read-only section
+ * that is physically placed at the 'lregion` argument.
+ *
+ */
+#define GROUP_ROM_LINK_IN(vregion, lregion) > lregion
+
 /*
  * As GROUP_LINK_IN(), but takes a second argument indicating the
  * memory region (e.g. "ROM") for the load address.  Used for
@@ -63,13 +71,15 @@
 #define GROUP_DATA_LINK_IN(vregion, lregion) > vregion
 #endif
 
-/*
- * The GROUP_FOLLOWS_AT() macro is located at the end of the section
- * and indicates that the section does not specify an address at which
- * it is to be loaded, but that it follows a section which did specify
- * such an address
+/**
+ * Route memory for read-write sections that are NOT loaded; typically this
+ * is only used for 'BSS' and 'noinit'.
  */
-#define GROUP_FOLLOWS_AT(where) AT > where
+#ifdef CONFIG_XIP
+#define GROUP_NOLOAD_LINK_IN(vregion, lregion) > vregion AT > vregion
+#else
+#define GROUP_NOLOAD_LINK_IN(vregion, lregion) > vregion
+#endif
 
 /*
  * The SECTION_PROLOGUE() macro is used to define the beginning of a section.

